File: Strings/Max_Chars/max_chars.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(maxChars("abcdadsfadgadsssssssa") == "s")
}

func maxChars(s string) string {

	charCounter := make(map[string]int)
	maxChar := ""
	maxCharCount := 0

	for _, c := range s {
		cStr := string(c)
		charCounter[cStr]++
		if charCounter[cStr] > maxCharCount {
			maxChar = cStr
			maxCharCount = charCounter[cStr]
		}
	}

	return maxChar
}

File: Strings/Max_Chars/max_chars.js
function maxChar(s) {
    const charCounter = {};
    let mostUsedChar = '';
    let mostUsedCharCount = 0;
    for (let i = 0; i < s.length; i++) {
        const c = s.charAt(i);
        if (!charCounter[c]) {
            charCounter[c] = 0;
        }
        charCounter[c]++;
        if (charCounter[c] > mostUsedCharCount) {
            mostUsedChar = c;
            mostUsedCharCount = charCounter[c];
        }
    }
    return mostUsedChar;
}

console.log(maxChar("abcdadsfadgadsssssssa") === 's');

